fulltext/controllers.py

```diff
@@ -1,105 +1,239 @@
+        # Pretend that the resource does not even exist.
+        raise NotFound('No such document')
+
+    try:
+        logger.debug('Create a new extraction task with %s, %s',
+                     identifier, id_type)
+        extract.create_task(identifier, id_type, owner, token)
+    except extract.TaskCreationFailed as e:
+        raise InternalServerError('Could not start extraction') from e
+    target = url_for('fulltext.task_status', identifier=identifier,
+                     id_type=id_type)
+    return ACCEPTED, status.ACCEPTED, {'Location': target}
+
+
+def get_task_status(identifier: str, id_type: str = SupportedBuckets.ARXIV,
+                    version: Optional[str] = None,
+                    authorizer: Optional[Authorizer] = None) -> Response:
     """Check the status of an extraction request."""
-    logger.debug('%s: Get status for task' % task_id)
-    if not isinstance(task_id, str):
-        logger.debug('%s: Failed, invalid task id' % task_id)
-        raise ValueError('task_id must be string, not %s' % type(task_id))
-    result = AsyncResult(task_id)
-    logger.debug('%s: got result: %s' % (task_id, result.status))
-    if result.status == 'PENDING':
-        return TASK_DOES_NOT_EXIST, HTTP_404_NOT_FOUND, {}
-    elif result.status in ['SENT', 'STARTED', 'RETRY']:
-        return TASK_IN_PROGRESS, HTTP_200_OK, {}
-    elif result.status == 'FAILURE':
-        logger.error('%s: failed task: %s' % (task_id, result.result))
-        reason = TASK_FAILED
-        reason.update({'reason': str(result.result)})
-        return reason, HTTP_200_OK, {}
-    elif result.status == 'SUCCESS':
-        task_result = result.result
-        document_id = task_result.get('document_id')
-        logger.debug('Retrieved result successfully, document_id: %s' %
-                     document_id)
-        headers = {'Location': url_for('fulltext.retrieve',
-                                       doc_id=document_id)}
-        return TASK_COMPLETE, HTTP_303_SEE_OTHER, headers
-    return TASK_DOES_NOT_EXIST, HTTP_404_NOT_FOUND, {}
+    logger.debug('get task status for %s in %s', identifier, id_type)
+    if id_type not in SupportedBuckets:
+        logger.debug('unsupported identifier type %s', id_type)
+        raise NotFound('Unsupported identifier')
+
+    storage = store.Storage.current_session()
+    try:
+        product = storage.retrieve(identifier, version, bucket=id_type,
+                                   meta_only=True)
+    except IOError as e:
+        logger.error('could not connect to storage backend')
+        raise InternalServerError('Could not connect to backend') from e
+    except store.DoesNotExist:
+        # If there is only metadata, we should still get a response from
+        # the store. So if we hit DoesNotExist there really is nothing to see
+        # here folks, move along.
+        logger.debug('store says does not exist')
+        raise NotFound('No such task')
```

Comment:
It appears that an unauthorized user can probe the existence of ID since they will get "no such task" for IDs that don't exist but "not authorized" for ones that do.

This might not be a concern.

Reply:
Oh, ignore that, I see they both raise NotFound('No such task')